### PR TITLE
[SYCL] Rename Plugin object -> Adapter.

### PR DESCRIPTION
**Pull request by aarongreig**
Also fix a few references to "plugin" in source files and a unittest script.

### CODE REVIEW

**Review by cppchedy — APPROVED** [review]

File: sycl/include/sycl/detail/ur.hpp [hunk 1]
```diff
@@ -103,8 +103,8 @@ __SYCL_EXPORT void contextSetExtendedDeleter(const sycl::context &constext,
                                              void *user_data);
 }
 
-class plugin;
-using PluginPtr = std::shared_ptr<plugin>;
+class adapter;
```
[the remark this answers is not in the document]
Comment (aarongreig) [reply sycl/include/sycl/detail/ur.hpp]: done, thought there might have been some weird collision motivating the original class being lower case but apparently not

**Review by Bensuo — APPROVED** [review]

**Review by sergey-semenov — APPROVED** [review]

**Comment by aarongreig** [comment]
ack, didn't notice there's still a required review never mind

**Review by kbenzie — APPROVED** [review]

File: sycl/source/detail/adapter.hpp [hunk 2]
```diff
@@ -1,4 +1,5 @@
-//==- plugin.hpp -----------------------------------------------------------==//
+//==- adapter.hpp
+//-----------------------------------------------------------==//
```
[the remark this answers is not in the document]
Comment (aarongreig) [reply sycl/source/detail/adapter.hpp]: fixed

**Review by RossBrunton — APPROVED** [review]

**Comment by aarongreig** [comment]
@intel/llvm-gatekeepers please merge

**Pull request merged**